### PR TITLE
chore: update to Rust 1.77.2

### PR DESCRIPTION
**Pull request by linbingquan**
update to Rust 1.77.1

If this PR has any problem, please tell me.

### CODE REVIEW

File: cli/util/fs.rs [hunk 1]
```diff
@@ -687,7 +688,7 @@ impl LaxSingleProcessFsFlag {
     let open_result = std::fs::OpenOptions::new()
       .read(true)
       .write(true)
-      .create(true)
+      .truncate(true)
```
Comment (mmastrac) [inline cli/util/fs.rs:691]: I have a feeling this isn't correct

Comment (linbingquan) [reply cli/util/fs.rs:691]: > I have a feeling this isn't correct

Thank you for your reply. 
Yes you are right.
Sorry my mistake...

```sh
warning: file opened with `create`, but `truncate` behavior not defined
   --> cli\util\fs.rs:691:8
    |
691 |       .create(true)
    |        ^^^^^^^^^^^^- help: add: `.truncate(true)`
```

**Comment by littledivy** [comment]
I tried this in https://github.com/denoland/deno/pull/23018 but clippy would crash in CI.

**Comment by linbingquan** [comment]
CI was failed in `lint debug linux-x86_64`, but I can't found file `ext/node/ops/vm_internal.rs`, maybe I miss something.

https://github.com/denoland/deno/actions/runs/8625000151/job/23640890407?pr=23262#step:29:1285

```sh
error: this `let...else` may be rewritten with the `?` operator
   --> ext/node/ops/vm_internal.rs:158:5
    |
158 | /     let Some(context) = object.get_creation_context(scope) else {
159 | |       return None;
160 | |     };
    | |______^ help: replace it with: `let context = object.get_creation_context(scope)?;`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#question_mark
    = note: `-D clippy::question-mark` implied by `-D clippy::all`
    = help: to override `-D clippy::all` add `#[allow(clippy::question_mark)]`
```

**Review by dsherret — APPROVED** [review]
LGTM too

**Pull request merged**